scripts/rebuildstaging.py

```diff
@@ -186,10 +185,9 @@ def rebuild_staging(config, print_details=True, push=True):
     merge_conflicts = []
     not_found = []
     all_configs = list(config.span_configs())
-    context_manager = contextlib.nested(*[OriginalBranch(get_git(path))
-                                          for path, _ in all_configs])
-    with context_manager:
+    with ExitStack() as stack:
         for path, config in all_configs:
+            stack.enter_context(OriginalBranch(get_git(path)))
```

Comment:
Not sure if it matters, but this is subtly different from what was done before. Here's an example that is closer to the previous code
```py
with ExitStack() as stack:
        for path, _ in all_configs:
            stack.enter_context(OriginalBranch(get_git(path)))
        for path, _ in all_configs:
            git = get_git(path)
            try:
                ...
```

Reply:
Yeah, I saw that - that's what I wrote at first actually, but then thought I'd try out just putting the `enter_context` inside the second loop and it seems to work fine.  Think it's problematic?  I don't think I fully grasp the implications, so happy to defer if this way seems safer.

Reply:
Seems fine if it works. However, if that works then I wonder if `ExitStack` could be eliminated by putting the context manager inside the for loop? Like this:
```py
for path, config in all_configs:
    git = get_git(path)
    with OriginalBranch(git):
        try:
            ...

Reply:
Good point - that makes more sense to me too.  Just tried it out and it seems to work fine.  Pushed that version.  Also tried it out with a merge conflict and still no issues